test/tests/functional/pbs_multi_sched.py

```diff
@@ -1644,34 +1645,42 @@ def test_auto_sched_off_due_to_fds_limit(self):
         when number of open files per process are exhausted
         """
 
-        # get the number of open files per process
-        ret = self.du.run_cmd(
-            self.server.hostname, [
-                'ulimit', '-n'], as_script=True, logerr=False)
-        self.assertEqual(ret['rc'], 0)
-        open_files_limit = ret['out'][0]
+        if os.getuid() != 0 or sys.platform in ('cygwin', 'win32'):
+            self.skipTest("Test need to run as root")
 
-        # set the number of open files per process to 10
-        ret = self.du.run_cmd(
-            self.server.hostname, [
-                'ulimit', '-n', '10'], as_script=True, logerr=False)
-        self.assertEqual(ret['rc'], 0)
+        try:
+            # get the number of open files per process
+            (open_files_soft_limit, open_files_hard_limit) =\
+                resource.getrlimit(resource.RLIMIT_NOFILE)
+
+            # set the soft limit of number of open files per process to 10
+            resource.setrlimit(resource.RLIMIT_NOFILE,
+                               (10, open_files_hard_limit))
+        except (ValueError, resource.error):
+            self.assertFalse("Error in accessing system RLIMIT_ variales, "
+                             "test fails.")
 
         self.setup_sc3()
 
-        for _ in range(0, 100):
-            self.server.manager(MGR_CMD_SET, SCHED,
-                                {'scheduling': 'True'}, id="sc3", expect=True)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduler_iteration': 1},
+                            id="sc3", expect=True)
+        self.server.manager(MGR_CMD_SET, SCHED, {'scheduling': 'True'},
+                            id="sc3", expect=True)
 
+        # This will trigger required number of scheduling cycles that are
+        # needed to exhaust open files per process which is 10 in our case.
+        time.sleep(15)
         # scheduling should not go to false once all fds per process
         # are exhausted.
         self.server.expect(SCHED, {'scheduling': 'True'},
                            id='sc3', max_attempts=10)
 
-        ret = self.du.run_cmd(
-            self.server.hostname, ['ulimit', '-n', open_files_limit],
-            as_script=True, logerr=False)
-        self.assertEqual(ret['rc'], 0)
+        try:
+            resource.setrlimit(resource.RLIMIT_NOFILE, (open_files_soft_limit,
+                                                        open_files_hard_limit))
+        except (ValueError, resource.error):
+            self.assertFalse("Error in accessing system RLIMIT_ variales, "
+                             "test fails.")
```

Comment:
Same as the above comment about the assert.

Reply:
Implemented.